### PR TITLE
Add strict initial objects

### PR DESCRIPTION
**Pull request by 4e554c4c**
This commit add strict initial object, and proves that any initial object in a CCC is strict. There is also a refactoring of the empty diagram.

### CODE REVIEW

File: src/Categories/Category/Construction/Empty.agda [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{-# OPTIONS --without-K --safe #-}
+
+module Categories.Category.Construction.Empty {o ℓ e}  where
```
Comment (Taneb) [inline src/Categories/Category/Construction/Empty.agda:3]: I think this already exists as `Categories.Category.Instance.Zero` (although your implementation is a lot tidier!)

Comment (4e554c4c) [reply src/Categories/Category/Construction/Empty.agda:3]: I see! I wonder why everything was using `liftC (Discrete 0)` then.

any preference on a name to use for both of them?

Comment (Taneb) [reply src/Categories/Category/Construction/Empty.agda:3]: Please stick with the existing name

Comment (4e554c4c) [reply src/Categories/Category/Construction/Empty.agda:3]: ah, there is also `Categories.Category.Instance.EmptySet`
quite a few of these.

Comment (Taneb) [reply src/Categories/Category/Construction/Empty.agda:3]: `Categories.Category.Instance.EmptySet` is certainly in the wrong place, but it's not a duplicate: it defines the initial objects for the `Sets` and `Setoids` categories

File: src/Categories/Object/StrictInitial.agda [hunk 2]
```diff
@@ -0,0 +1,30 @@
+{-# OPTIONS --without-K --safe #-}
+open import Categories.Category
```
Comment (JacquesCarette) [inline src/Categories/Object/StrictInitial.agda:2]: `.Core using (Category)`

**Review by JacquesCarette — CHANGES_REQUESTED** [review]
Beautiful refactoring! One minor thing to fix before I merge.

**Comment by JacquesCarette** [comment]
On the question

> I wonder why everything was using liftC (Discrete 0) then

This is because `Data.Empty.Polymorphic` did not yet exist when this was created. I contributed those because of this lack, but never had the chance to come back to `agda-categories` to refactor!

**Review by JacquesCarette — APPROVED** [review]

**Pull request merged**